src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -81,54 +79,50 @@ protected virtual void OnScriptFileChanged(object sender, FileSystemEventArgs e)
         {
         }
 
-        protected static Dictionary<string, string> GetBindingData(object value, IBinderEx binder, Collection<FunctionBinding> inputBindings, Collection<FunctionBinding> outputBindings)
+        /// <summary>
+        /// Get the binding data. In dynamic script cases we need
+        /// to parse this POCO data ourselves - it won't be in the existing
+        /// binding data because all the POCO binders require strong
+        /// typing
+        /// </summary>
+        protected static Dictionary<string, string> GetBindingData(object value, IBinderEx binder)
         {
+            // First apply any existing binding data. Any additional binding
+            // data coming from the message will take precedence
             Dictionary<string, string> bindingData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            ApplyAmbientBindingData(binder, bindingData);
 
-            // If there are any parameters in the bindings,
-            // get the binding data. In dynamic script cases we need
-            // to parse this POCO data ourselves - it won't be in the existing
-            // binding data because all the POCO binders require strong
-            // typing
-            if (outputBindings.Any(p => p.HasBindingParameters) ||
```

Comment:
Removing this check (i.e. always extract binding data) is what allowed me to remove the FunctionBinding.HasBindingParameters property.